src/encoded/tests/test_metadata.py

```diff
@@ -0,0 +1,1565 @@
+        ('field', 'replicates.library.biosample.treatments.treatment_term_name'),
+        ('field', 'replicates.library.biosample.treatments.amount'),
+        ('field', 'replicates.library.biosample.treatments.amount_units'),
+        ('field', 'replicates.library.biosample.treatments.duration'),
+        ('field', 'replicates.library.biosample.treatments.duration_units'),
+        ('field', 'replicates.library.biosample.applied_modifications.method'),
+        ('field', 'replicates.library.biosample.applied_modifications.category'),
+        ('field', 'replicates.library.biosample.applied_modifications.modified_site_by_target_id'),
+        ('field', 'replicates.library.biosample.applied_modifications.modified_site_by_gene_id'),
+        ('field', 'replicates.library.biosample.applied_modifications.modified_site_by_coordinates.assembly'),
+        ('field', 'replicates.library.biosample.applied_modifications.modified_site_by_coordinates.chromosome'),
+        ('field', 'replicates.library.biosample.applied_modifications.modified_site_by_coordinates.start'),
+        ('field', 'replicates.library.biosample.applied_modifications.modified_site_by_coordinates.end'),
+        ('field', 'replicates.library.biosample.applied_modifications.zygosity'),
+        ('field', 'target.name'),
+        ('field', 'replicates.library.nucleic_acid_term_name'),
+        ('field', 'replicates.library.depleted_in_term_name'),
+        ('field', 'replicates.library.extraction_method'),
+        ('field', 'replicates.library.lysis_method'),
+        ('field', 'replicates.library.crosslinking_method'),
+        ('field', 'replicates.library.strand_specificity'),
+        ('field', 'date_released'),
+        ('field', 'award.project'),
+        ('field', 'files.replicate.rbns_protein_concentration'),
+        ('field', 'files.replicate.rbns_protein_concentration_units'),
+        ('field', 'files.replicate.library.fragmentation_method'),
+        ('field', 'files.replicate.library.size_range'),
+        ('field', 'files.biological_replicates'),
+        ('field', 'files.technical_replicates'),
+        ('field', 'files.read_length'),
+        ('field', 'files.mapped_read_length'),
+        ('field', 'files.run_type'),
+        ('field', 'files.paired_end'),
+        ('field', 'files.paired_with'),
+        ('field', 'files.index_of'),
+        ('field', 'files.derived_from'),
+        ('field', 'files.file_size'),
+        ('field', 'files.lab.title'),
+        ('field', 'files.md5sum'),
+        ('field', 'files.dbxrefs'),
+        ('field', 'files.href'),
+        ('field', 'files.genome_annotation'),
+        ('field', 'files.platform.title'),
+        ('field', 'files.controlled_by'),
+        ('field', 'files.status'),
+        ('field', 'files.no_file_available'),
+        ('field', 'files.restricted'),
+        ('field', 'files.s3_uri')
+    ]
+    for param in mr._get_field_params():
+        assert param in expected_field_params
```

Comment:
I’m not sure how useful this test is, but it does seem more useful than the one it replaces, which compared two static lists to see we copied and pasted them correctly.